pkg/kn/traffic/compute.go

```diff
@@ -197,53 +197,55 @@ func errorOverWritingTag(tag string) error {
 		"add flag '--untag %s' in command to untag it", tag)
 }
 
-func errorRepeatingLatestRevision(forFlag string) error {
+func errorRepeatingRevision(forFlag string, name string) error {
 	return fmt.Errorf("repetition of identifier %s "+
-		"is not allowed, use only once with %s flag", latestRevisionRef, forFlag)
+		"is not allowed, use only once with %s flag", name, forFlag)
 }
 
 // verifies if user has repeated @latest field in --tag or --traffic flags
 // verifyInputSanity checks:
 // - if user has repeated @latest field in --tag or --traffic flags
 // - if provided traffic portion are integers
 func verifyInputSanity(trafficFlags *flags.Traffic) error {
-	var latestRevisionTag = false
-	var latestRevisionTraffic = false
 	var sum = 0
 
-	for _, each := range trafficFlags.RevisionsTags {
+	// To heck if there are duplicated @latest in --tag flags
+	revisionMap := make(map[string]int)
+	for i, each := range trafficFlags.RevisionsTags {
 		revision, _, err := splitByEqualSign(each)
 		if err != nil {
 			return err
 		}
 
-		if latestRevisionTag && revision == latestRevisionRef {
-			return errorRepeatingLatestRevision("--tag")
-
-		}
-
-		if revision == latestRevisionRef {
-			latestRevisionTag = true
+		fmt.Println("---", revision)
+		if _, exist := revisionMap[revision]; exist && revision == latestRevisionRef {
+			return errorRepeatingRevision("--tag", latestRevisionRef)
+		} else {
+			revisionMap[revision] = i
 		}
 	}
 
-	for _, each := range trafficFlags.RevisionsPercentages {
+	revisionRefMap := make(map[string]int)
+	for i, each := range trafficFlags.RevisionsPercentages {
 		revisionRef, percent, err := splitByEqualSign(each)
 		if err != nil {
 			return err
 		}
 
+		// To heck if there are duplicated revision names in traffic flags
+		if _, exist := revisionRefMap[revisionRef]; exist {
+			return errorRepeatingRevision("--traffic", revisionRef)
```

Comment:
a user can repeat the revisions in traffic block

Reply:
I should say revisionRef here as user can provide revision as well tag in `--traffic` block